### PR TITLE
ESP32 RMT-fix: Need to clear struct before filing it out.

### PR DESCRIPTION
**Pull request by mishafarms**
Signed-off-by: mishafarms <github@mishafarms.us>

### CODE REVIEW

**Comment by mishafarms** [comment]
The Issue was that I was getting a 1Mhz clock instead of the 80Mhz clock. This was because the flags field is not filled in and when allocated on the stack can have garbage in it.

File: ports/esp32/esp32_rmt.c [hunk 1]
```diff
@@ -96,7 +96,9 @@ STATIC mp_obj_t esp32_rmt_make_new(const mp_obj_type_t *type, size_t n_args, siz
     self->carrier_freq = carrier_freq;
     self->loop_en = false;
 
-    rmt_config_t config;
+    // start with an empty config, not what was on the stack
+
+    rmt_config_t config = {};
```
Comment (dpgeorge) [inline ports/esp32/esp32_rmt.c:101]: Other parts of the code base use `= {0}` for this, so would be good to do that here as well, for consistency.

Also, IMO the comment is not needed, if you know C it's pretty clear what this is doing.

Comment (mishafarms) [reply ports/esp32/esp32_rmt.c]: Done.

**Review by dpgeorge — APPROVED** [review]

**Comment by dpgeorge** [comment]
Thanks for this.  Merged in 8ff3520f67bb8f7e5509129f9f561a1217ababa4

**Pull request closed without merging**